### PR TITLE
fix: core link styles a11y

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Links and inline buttons should have the underline style

## What is the new behavior?
Links and buttons have the proper underline style. Also removed a trailing white space that caused a bit of excess underline in the inline button.

## Does this PR introduce a breaking change?
minor visual but a fix for a11y
- [ ] Yes
- [x] No (ish)

### CODE REVIEW

File: packages/core/src/button/inline-button.element.ts [hunk 1]
```diff
@@ -22,11 +22,11 @@ import { styles } from './inline-button.element.css.js';
  * @beta
  * @element cds-inline-button
  * @slot - Content slot for inside the button
+ * @cssprop --text-decoration
  * @cssprop --color
  * @cssprop --font-size
- * @cssprop --font-weight
+ * @cssprop --line-height
```
Comment (mathisscott) [inline packages/core/src/button/inline-button.element.ts:28]: Why the changes here?

**Comment by mathisscott** [comment]
This change should only go into master/v5. It's technically a visual breaking change.

File: packages/core/src/button/inline-button.element.ts (hunk 1 above)
Comment (coryrylan) [reply packages/core/src/button/inline-button.element.ts:28]: They didnt match the properties being used in the component https://github.com/vmware/clarity/blob/master/packages/core/src/button/inline-button.element.scss#L9
